### DISCUSSION TITLE
Shadow driving me crazy

### DISCUSSION THREAD

**Discussion by hermanfenderson** [ImageMagick/ImageMagick#2772] [Help]
This is driving me crazy.
I got a file prova.jpg  and this transformation 
magick convert prova.jpg -resize 800x900 ( +clone -background black -shadow 70x10+5+5 ) +swap -background "rgb(242,242,242)" -layers merge -background "rgb(242,242,242)" -compose Copy -gravity center -geometry +20+20 -extent 900x1000 -quality 92 prova7.jpg

![prova](https://user-images.githubusercontent.com/462416/97498057-cc9d7b80-196b-11eb-8c15-c9598213f98b.jpg)
It should give me a nice shadow and resize the image...

Matter of fact I get this:
![prova7](https://user-images.githubusercontent.com/462416/97498113-e343d280-196b-11eb-83c1-fb435310564f.jpg)

But... if I simpy open the original jpg in paint and "save as" prova2.jpg...
![prova2](https://user-images.githubusercontent.com/462416/97498159-fb1b5680-196b-11eb-9c4c-ad0b084ab7dc.jpg)

I got the result I was expecting with the very same transformation.

![prova72](https://user-images.githubusercontent.com/462416/97498186-066e8200-196c-11eb-8d94-424519aee500.jpg)

Cannot find out the problem!!! And it is driving me crazy.

**Comment by fmw42** [comment]
Try this simpler command. Note that on IM 7 use magick, not magick convert.

```
magick  prova.jpg -resize 800x900 ^
( +clone -background "gray(242)" -shadow 70x10+5+5 ) ^
+swap -layers merge +repage ^
-gravity center -background "gray(242)" -extent 900x1000+20+20 ^
-quality 92 prova7.jpg
```

![prova7](https://user-images.githubusercontent.com/13458218/97499308-e5d20800-192a-11eb-9717-943b83fccd02.jpg)

**Reply by hermanfenderson** [reply]
Works perfectly. Black magic to me. But tnx. You saved my day!!!

**Reply by fmw42** [reply]
I basically added the shadow (but with your gray color and not black) and then extended the canvas using -extent to the final size you wanted. That avoided another composite.

**Reply by hermanfenderson** [reply]
Ok...
But seems anyhow very picky.
Look at this example...

![prova](https://user-images.githubusercontent.com/462416/97500452-e345d180-196f-11eb-9ad3-57f409e1a172.jpg)

Should give me the same result (this is the input) ... but the shadow in output is gone... 

![prova7](https://user-images.githubusercontent.com/462416/97500614-2f911180-1970-11eb-8670-30b5906c4d6a.jpg)

**Reply by hermanfenderson** [reply]
The funny part being that my original command is working on this...
Cannot understand why.

![prova7](https://user-images.githubusercontent.com/462416/97503781-f9569080-1975-11eb-9c9f-a40e5dea33b1.jpg)

**Reply by fmw42** [reply]
The reason is that your original image was CMYK and your second image is sRGB. With sRGB you need to make your shadow darker. Oddly, with CMYK, which is not how it should be done, you get a better shadow (more faded and rounded). But the trick is to convert your sRGB image to CMYK, do the processing and then convert back to sRGB. So the following works for me:

```
magick test.jpg -resize 800x900 -colorspace CMYK ^
( +clone -background "gray(242)" -shadow 70x10+5+5 ) ^
+swap -layers merge +repage ^
-gravity center -background "gray(242)" -extent 900x1000+20+20 ^
-colorspace sRGB -quality 92 test7.jpg
```

![test7](https://user-images.githubusercontent.com/13458218/97504135-862c2a80-1933-11eb-903d-559c00e774c6.jpg)

and your original image, which is CMYK already, just needs to be converted at the end to sRGB, so that the shadow is gray and not some odd brownish color from CMYK.

```
magick prova.jpg -resize 800x900 ^
( +clone -background "gray(242)" -shadow 70x10+5+5 ) ^
+swap -layers merge +repage ^
-gravity center -background "gray(242)" -extent 900x1000+20+20 ^
-colorspace sRGB -quality 92 prova7.jpg
```
![prova7](https://user-images.githubusercontent.com/13458218/97504285-cee3e380-1933-11eb-9628-734d259d8248.jpg)

**Reply by hermanfenderson** [reply]
I need to learn a lot!
Tnx 1+E06!!!!
